majilove/models.py

```diff
@@ -492,3 +493,205 @@ def __unicode__(self):
                 name=self.t_name,
             )
         return u''
+
+############################################################
+# Collectible Photos
+
+class CollectiblePhoto(AccountAsOwnerModel):
+    collection_name = 'collectiblephoto'
+
+    account = models.ForeignKey(Account, verbose_name=_('Account'), related_name='photoscollectors')
+    photo = models.ForeignKey(Photo, verbose_name=_('Photo'), related_name='collectedphotos')
+    level = models.PositiveIntegerField(_('Level'), default=1)
+    leader_bonus = models.PositiveIntegerField(_('Leader skill percentage'), null=True)
+    skill_level = models.PositiveIntegerField(_('Skill level'), default=1)
+
+    @property
+    def skill_percentage(self):
+        return self.photo.skill_percentage + (self.skill_level - 1) * self.photo.skill_increment
+    @property
+    def skill_note_count(self):
+        return self.photo.skill_note_count + (self.skill_level - 1) * self.photo.skill_increment
+
+    @property
+    def skill(self):
+        return self.photo.skill_template.format({
+            k: getattr(self, k)
+            for k in templateVariables(self.photo.skill_template)
+        })
+
+    sub_skill_level = models.PositiveIntegerField(_('Sub skill level'), null=True)
+    @property
+    def sub_skill_amount(self):
+        return self.photo.sub_skill_amount + (self.sub_skill_level - 1) * self.photo.sub_skill_increment
+
+    @property
+    def sub_skill(self):
+        _sub_skill_variables = {k: getattr(self.photo, k)
+        for k in templateVariables(self.photo.sub_skill_template
+        )}
+        _sub_skill_variables['sub_skill_amount'] = self.sub_skill_amount
+        return self.photo.sub_skill_template.format(**_sub_skill_variables)
+
+    rank = models.PositiveIntegerField(_('Rank'), default=1)
+
+    # TODO: moment based things are the same across, do I need an intermediate model for moment things?
+
+    # percentage displayed on moments page for the card
+    moments_unlocked = models.PositiveIntegerField(_('Percent of moments unlocked'), default = 0)
+    # integer number of squares
+    bonus_moment_squares_unlocked = models.PositiveIntegerField(_('Number of moment squares unlocked past 100%'), default = 0)
+    @property
+    def special_shot_unlocked(self):
+        return self.moments_unlocked >= self.photo.special_shot_percentage
+
+    prefer_normal_shot = models.BooleanField(_('Prefer normal shot photo image'), default=False)
+
+    @property
+    def image(self):
+        return self.photo.image_special_shot if self.special_shot_unlocked and not self.prefer_normal_shot else self.photo.image
+
+    @property
+    def art(self):
+        return self.photo.art_special_shot if self.special_shot_unlocked and not self.prefer_normal_shot else self.photo.art
```

Comment:
You may also need `art_url`, `http_art_url`, `image_url` and `http_image_url` in the future.